src/MatchingResult.php

```diff
@@ -9,14 +9,14 @@
 
 final class MatchingResult implements MiddlewareInterface
 {
-    private $success;
+    private ?bool $success;
 
     /**
      * @var Route
      */
-    private $route;
-    private $parameters = [];
-    private $methods;
+    private Route $route;
```

Comment:
Is there a reason not to initialize it?

Reply:
1. For some reason (I don't know why) it is not initialized when creating in the `fromFailure` method. But there will be an error if smbdy will try to `process` this object without `route` property.
1. This property is filled with some some object when creating a `MatcingResult` with `fromSuccess`. And we will spend some extra time and memory if we will fill it with wome another value before this.

Conclusion: this property must be filled, but I don't know the best place and the best way to od this.

Reply:
Re-checked. This one is OK as is. It's used in `process()` and there's a check necessary to prevent error.

Reply:
Made a check there. Look at it please.